blobfuse/src/BlobStreamer.cpp

```diff
@@ -95,7 +97,30 @@ BlobBlock* BlobStreamer::GetBlock(const char* file_name, uint64_t offset, Stream
         block->valid = true;
         block->last = false;
 
-        azclient->DownloadToStream(file_name, block->buff, start_offset, block_size);
+        uint64_t download_size = block_size;
+        if (download_size > MAX_BLOCK_SIZE_FOR_SINGLE_READ && 
+            obj->GetSize() > 0) 
+        {
+            if ((start_offset + download_size) > obj->GetSize()) {
+                download_size = obj->GetSize() - start_offset;
+                if (download_size == 0) {
+                    obj->UnLock();
+                    syslog(LOG_ERR, "Failed to download block of %s with offset %lu.  Errno : (Out of range).\n", file_name, start_offset);
+                    errno = 416;
+                    return NULL;
+                }
+            }
+        }
+
+        if (download_size < MAX_BLOCK_SIZE_FOR_SINGLE_READ) {
+            azclient->DownloadToStream(file_name, block->buff, start_offset, download_size);
+        } else {
+            char *buff = (char*)malloc(download_size);
+            azclient->DownloadToBuffer(file_name, buff, start_offset, download_size, config_options.concurrency);
```

Comment:
I think we should explain to the user in the README how max-concurrency attribute will be used in the streaming case? since this is ignored if the block size is smaller than 64MB.

Reply:
Updated readme for the same.